lib/dry/configurable/settings.rb

```diff
@@ -23,6 +23,35 @@ def initialize(elements = EMPTY_ARRAY)
         initialize_elements(elements)
       end
 
+      def replace(settings)
+        unless settings.is_a? Dry::Configurable::Settings
+          raise ArgumentError, "settings must be a Dry::Configurable::Settings"
+        end
+
+        settings.each do |setting|
+          self << setting
+        end
+      end
+
+      def merge(settings)
+        unless settings.is_a? Dry::Configurable::Settings
+          raise ArgumentError, "settings must be a Dry::Configurable::Settings"
+        end
+
+        settings.each do |setting|
+          merge_setting(setting)
+        end
+      end
+
+      # @api private
+      def merge_setting(setting)
```

Comment:
BTW, this method should be moved below the `private` line (and we can then ditch the `# @api private` comment at the same time).